.github/workflows/pr.yml

```
@@ -0,0 +1,11 @@
+name: 'PR checks'
+on:
+  workflow_dispatch:
+  pull_request:
+    types: [opened, synchronize]
+
+jobs:
+  build-and-test:
+    uses: ./.github/workflows/build-and-test.yml
+    with:
+      build-all-platforms: false
```

we can explicitly add the default minimal permissions for clarity:

```suggestion
name: 'PR checks'
on:
  workflow_dispatch:
  pull_request:
    types: [opened, synchronize]

permissions:
  contents: read

jobs:
  build-and-test:
    uses: ./.github/workflows/build-and-test.yml
    with:
      build-all-platforms: false
    permissions:
      contents: read
```